components/tidb_query_codegen/src/rpn_function.rs

```diff
@@ -1129,21 +1220,39 @@ impl NormalRpnFn {
         let mut arg_types = Vec::new();
         let mut arg_types_anonymous = Vec::new();
         let mut arg_types_no_ref = Vec::new();
-        for fn_arg in item_fn.sig.inputs.iter().skip(attr.captures.len()) {
+        let take_cnt =
+            item_fn.sig.inputs.len() - attr.captures.len() - if attr.writer { 1 } else { 0 };
```

Comment:
as usize?

Reply:
Thanks for reviewing! I've changed this part.